wso2.org.component.javaparser/src/main/java/component/javaparser/WSO2JavaParser.java

```diff
@@ -0,0 +1,265 @@
+        Matcher ds = cs.matcher(reference);
+        while (ds.find()) {
+            scrReference.setPolicy(ds.group(1));
+            // System.out.println("Policy: " + scrReference.getPolicy());
+        }
+        Pattern csd = Pattern.compile("unbind=\"([^\"]*)\"");
+        Matcher dsd = csd.matcher(reference);
+        while (dsd.find()) {
+            scrReference.setUnbind(dsd.group(1));
+            //System.out.println("unbind: " + scrReference.getUnbind());
+        }
+        Pattern csdx = Pattern.compile("bind=\"([^\"]*)\"");
+        String dsq = reference.replace("unbind", "***");
+        Matcher dsdx = csdx.matcher(dsq);
+        while (dsdx.find()) {
+            scrReference.setBind(dsdx.group(1));
+            // System.out.println("bind: " + scrReference.getBind());
+        }
+        System.out.println("\n");
+        return scrReference;
+    }
+
+    // This method get the java files from internal directory
+    public List<String> getInternalFiles(String directoryPath) {
+        File folder = new File(directoryPath);
+        File[] listOfFiles = folder.listFiles();
+        List<String> fileNames = new ArrayList<>();
+        if (listOfFiles != null) {
+            for (int i = 0; i < listOfFiles.length; i++) {
+                if (listOfFiles[i].isFile()) {
+                    String file = directoryPath + "/" + listOfFiles[i].getName();
+                    fileNames.add(file);
+                }
+            }
+        }
+        return fileNames;
+    }
+
+    public List<String> getComponentPaths(String file) {
+        List<String> paths = new ArrayList<>();
+        Document doc = readXMLDoc(file);
+        NodeList nodeList = doc.getElementsByTagName("Path");
+        for (int temp = 0; temp < nodeList.getLength(); temp++) {
+            paths.add(nodeList.item(temp).getTextContent());
+            //System.out.println(nodeList.item(temp).getTextContent());
+        }
+
+        return paths;
+    }
+
+    public List<String> getPomPaths(String file) {
+        List<String> pomPaths = new ArrayList<>();
+        Document doc = readXMLDoc(file);
+        NodeList nodeList = doc.getElementsByTagName("POMPath");
+        for (int temp = 0; temp < nodeList.getLength(); temp++) {
+            pomPaths.add(nodeList.item(temp).getTextContent());
+        }
+        return pomPaths;
+    }
+
+    public Document readXMLDoc(String file) {
+        File inputFile = new File(file);
+        DocumentBuilderFactory dbFactory = DocumentBuilderFactory.newInstance();
+        DocumentBuilder dBuilder = null;
+        try {
+            dBuilder = dbFactory.newDocumentBuilder();
+        } catch (ParserConfigurationException e) {
+            e.printStackTrace();
+        }
+        Document doc = null;
+        try {
+            doc = dBuilder.parse(inputFile);
+        } catch (SAXException e) {
+            e.printStackTrace();
+        } catch (IOException e) {
+            e.printStackTrace();
```

Comment:
did we use a logging mechanism? I do not think we need one for this though.

Reply:
No. I removed this.